jami-android/app/src/main/java/cx/ring/service/RemoteControl.kt

```diff
@@ -0,0 +1,224 @@
+            .setPriority(NotificationCompat.PRIORITY_LOW)
+            .setOngoing(true)
+            .build()
+
+        startForeground(NOTIFICATION_ID, notification)
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        Log.d(tag, "Service destroyed")
+        compositeDisposable.clear()
+    }
+
+    // Inner Binder class implementing the AIDL interface
+    private val binder = object : IRemoteService.Stub() {
+
+        override fun createAccount(input: String): String {
+            Log.d(tag, "Creating account with login: $input")
+            return try {
+                val map = fillMap(AccountData.data, input)
+                Log.d(tag, "Generated account data map: $map")
+
+                val accountId = accountService.addAccount(map)
+                    .map { account -> account.accountId }
+                    .doOnError { error -> Log.e(tag, "Error adding account: ${error.message}") }
+                    .blockingFirst()
+
+                Log.d(tag, "Account created successfully with ID: $accountId")
+                accountId
+            } catch (e: Exception) {
+                Log.e(tag, "Failed to create account: ${e.message}", e)
+                throw RuntimeException("Account creation failed", e)
+            }
+        }
+
+        override fun getAccountId(): String {
+            Log.d(tag, "Fetching account ID")
+            return try {
+                accountService.currentAccount?.accountId ?: ""
+            } catch (e: Exception) {
+                Log.e(tag, "Failed to fetch account ID: ${e.message}", e)
+                ""
+            }
+        }
+
+        override fun addContact(contactId: String) {
+            try {
+                val accountId = accountService.currentAccount?.accountId
+                    ?: throw IllegalStateException("No account found")
+                accountService.addContact(accountId, contactId)
+                Log.d(tag, "Successfully added contact: $contactId to account: $accountId")
+            } catch (e: Exception) {
+                Log.e(tag, "Failed to add contact: $contactId to account: $accountId", e)
+                throw RemoteException("Failed to add contact: ${e.message}")
+            }
+        }
+
+        override fun isContactExist(contactId: String): Boolean {
+            try {
+                val contacts : Map<String, Contact>? = accountService.currentAccount?.contacts
+                return contacts?.containsKey(contactId) ?: false
+            } catch (e: Exception) {
+                Log.e(tag, "Failed to check contact: $contactId", e)
+                throw RemoteException("Failed to check contact: ${e.message}")
+            }
+        }
+
+        override fun sendTrustRequest(contactId: String) {
+            try {
+                val accountId = accountService.currentAccount?.accountId
+                    ?: throw IllegalStateException("No account found")
```

Comment:
Does this exception end up back at the caller?

Reply:
Yes, if isContactExist throws a RemoteException, it will be sent back to the client because AIDL is designed to propagate such exceptions. It will end up here:

```kotlin
        lifecycleScope.launch(card.cardJob) {
            try {
/..................................................................................................../
                                    jamiService?.let { service: IRemoteService ->
                                        if (!service.isContactExist(jamiId)) {
                                            Timber.tag(TAG).d("Adding Jami contact: $jamiId")
                                            service.addContact(jamiId)
                                            service.sendTrustRequest(jamiId)
                                        } else {
                                            Timber.tag(TAG).d("Jami contact already exists: $jamiId")
                                        }
                                        callJami(jamiId)
                                    }
                                }

                            }
                        }
                    }
  /..................................................................................................../
                    }
                }
            } catch (e: Exception) {  <---- we should catch it here
                releaseSource()
                crashlytics.recordException(e)
            }
```

Reply:
Perfect.